src/state.cpp

```diff
@@ -161,8 +165,9 @@ State::external_join(const bytes& leaf_secret,
 {
   auto initial_state = State(std::move(sig_priv), pgs);
   auto add = initial_state.add_proposal(kp);
+  auto opts = CommitOpts{ { add }, true };
```

Comment:
```suggestion
  auto opts = CommitOpts{ { add }, false };
```

Nobody is going to use the Welcome here, so we shouldn't incur the expense of serializing out the tree.